cmd/kueuectl/app/list/list_localqueue.go

```diff
@@ -0,0 +1,226 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/printers"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned"
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	kueuev1beta1 "sigs.k8s.io/kueue/client-go/clientset/versioned/typed/kueue/v1beta1"
+)
+
+const (
+	lqLong = `Lists LocalQueues that match the given criteria: point to a specific CQ, 
+being active/inactive, belonging to the specified namespace or matching 
+the label selector.`
+	lqExample = `  # List LocalQueue 
+  kueuectl list localqueue`
+)
+
+type LocalQueueOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	AllNamespaces      bool
+	EnforceNamespace   bool
+	Namespace          string
+	FieldSelector      string
+	LabelSelector      string
+	ClusterQueueFilter string
+
+	Client kueuev1beta1.KueueV1beta1Interface
+
+	TablePrintObj   printers.ResourcePrinterFunc
+	DefaultPrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewLocalQueueOptions(streams genericiooptions.IOStreams) *LocalQueueOptions {
+	return &LocalQueueOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewLocalQueueCmd(clientGetter genericclioptions.RESTClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewLocalQueueOptions(streams)
+
+	cmd := &cobra.Command{
+		Use: "localqueue [-–clusterqueue CLUSTER_QUEUE_NAME] [--selector key1=value1] [--field-selector key1=value1] [--all-namespaces]",
+		// To do not add "[flags]" suffix on the end of usage line
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"lq"},
+		Short:                 "List LocalQueue",
+		Long:                  lqLong,
+		Example:               lqExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Validate())
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addClusterQueueFilterFlagVar(cmd, &o.ClusterQueueFilter)
```

Comment:
Can we have a set of flags show up as named flags group?

Reply:
What do you mean?

Reply:
When you type `<> help list localqueue` to get in the output something like :

```
filter-flags:
  selector: .....
  field-selector: .....
```

Reply:
It's possible to group commands, but I'm not sure about flags.

Reply:
I propose do not use it because it is not add groups on printing.

Reply:
Fine by me